docs/book/src/migration/v2vsv3.md

```diff
@@ -0,0 +1,16 @@
+# Kubebuilder v2 vs v3
+
+This document cover all breaking changes when migrating from v2 to v3.
+
+The details of all changes (breaking or otherwise) can be found in [kubebuilder](https://github.com/kubernetes-sigs/kubebuilder/releases) release notes.
+
+## Kubebuilder
+
+- V1 projects were deprecated and are no longer supported.
+- The deprecated args related to Go modules such as `--dep` and `depArgs` were removed
```

Comment:
```suggestion
```
This is not part of the v2->v3 migration, they are already deprecated in v2, so I would not mention it here.

Reply:
I think it is fair enough 👍